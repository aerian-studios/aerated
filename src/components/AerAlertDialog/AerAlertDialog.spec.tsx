import React from "react";
import { describe, it, expect, vi, afterEach } from "vitest";
import { cleanup, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { AerAlertDialog, AerAlertDialogFooter } from "./index";

describe("AlertDialog", () => {
  afterEach(() => {
    cleanup();
  });

  it("should render correctly", async () => {
    const { baseElement } = render(
      <AerAlertDialog
        trigger={<button>Open the dialog</button>}
        dialogTitle={<>This is important!</>}
        footer={
          <AerAlertDialogFooter
            cancel={<button>Cancel</button>}
            action={<button>Yes, do it!</button>}
          />
        }
        content={<p>Are you sure about all of that stuff?</p>}
      />
    );

    await userEvent.click(
      screen.getByRole("button", { name: "Open the dialog" })
    );

    expect(screen.getByRole("alertdialog", { name: "This is important!" }));
    expect(screen.getByText("Are you sure about all of that stuff?"));

    expect(baseElement).toMatchSnapshot();
  });

  it("should render an accessible `alertdialog` despite not showing the title visually", async () => {
    const { baseElement } = render(
      <AerAlertDialog
        trigger={<button>Open the dialog</button>}
        dialogTitle={{ title: <>This is important!</>, hideTitle: true }}
        footer={
          <AerAlertDialogFooter
            cancel={<button>Cancel</button>}
            action={<button>Yes, do it!</button>}
          />
        }
        content={<p>Are you sure about all of that stuff?</p>}
      />
    );

    await userEvent.click(
      screen.getByRole("button", { name: "Open the dialog" })
    );

    expect(screen.getByRole("alertdialog", { name: "This is important!" }));
    expect(baseElement).toMatchSnapshot();
  });

  it("should call the cancel callback and hide the dialog", async () => {
    const cancelCb = vi.fn();

    const { baseElement } = render(
      <AerAlertDialog
        trigger={<button>Open the dialog</button>}
        dialogTitle={<>This is important!</>}
        footer={
          <AerAlertDialogFooter
            cancel={<button onClick={cancelCb}>Cancel</button>}
            action={<button>Yes, do it!</button>}
          />
        }
        content={<p>Are you sure about all of that stuff?</p>}
      />
    );

    await userEvent.click(
      screen.getByRole("button", { name: "Open the dialog" })
    );

    await userEvent.click(screen.getByRole("button", { name: /cancel/i }));

    expect(cancelCb).toHaveBeenCalledOnce();
  });

  it("should call the action callback and hide the dialog", async () => {
    const actionCb = vi.fn();

    const { baseElement } = render(
      <AerAlertDialog
        trigger={<button>Open the dialog</button>}
        dialogTitle={"This is important!"}
        footer={
          <AerAlertDialogFooter
            cancel={<button>Cancel</button>}
            action={<button onClick={actionCb}>Yes, do it!</button>}
          />
        }
        content={<p>Are you sure about all of that stuff?</p>}
      />
    );

    await userEvent.click(
      screen.getByRole("button", { name: "Open the dialog" })
    );

    await userEvent.click(screen.getByRole("button", { name: /yes, do it/i }));

    expect(actionCb).toHaveBeenCalledOnce();
  });
});
