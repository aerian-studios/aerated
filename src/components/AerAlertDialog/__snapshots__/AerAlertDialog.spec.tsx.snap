// Vitest Snapshot v1

exports[`AlertDialog > should render an accessible \`alertdialog\` despite not showing the title visually 1`] = `
<body
  style="pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <button
      aria-controls="radix-:r3:"
      aria-expanded="true"
      aria-haspopup="dialog"
      data-state="open"
      type="button"
    >
      Open the dialog
    </button>
  </div>
  <div
    class=""
  >
    <div
      aria-hidden="true"
      class="_overlay_e6bcc0"
      data-aria-hidden="true"
      data-state="open"
      style="pointer-events: auto;"
    />
    <div
      aria-describedby="radix-:r5:"
      aria-labelledby="radix-:r4:"
      class="_content_e6bcc0"
      data-state="open"
      id="radix-:r3:"
      role="alertdialog"
      style="pointer-events: auto;"
      tabindex="-1"
    >
      <h2
        class="_title_e6bcc0"
        id="radix-:r4:"
      >
        This is important!
      </h2>
      <p
        class="_description_e6bcc0"
        id="radix-:r5:"
      >
        Are you sure about all of that stuff?
      </p>
      <footer>
        <button
          type="button"
        >
          Cancel
        </button>
        <button
          type="button"
        >
          Yes, do it!
        </button>
      </footer>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;

exports[`AlertDialog > should render correctly 1`] = `
<body
  style="pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <button
      aria-controls="radix-:r0:"
      aria-expanded="true"
      aria-haspopup="dialog"
      data-state="open"
      type="button"
    >
      Open the dialog
    </button>
  </div>
  <div
    class=""
  >
    <div
      aria-hidden="true"
      class="_overlay_e6bcc0"
      data-aria-hidden="true"
      data-state="open"
      style="pointer-events: auto;"
    />
    <div
      aria-describedby="radix-:r2:"
      aria-labelledby="radix-:r1:"
      class="_content_e6bcc0"
      data-state="open"
      id="radix-:r0:"
      role="alertdialog"
      style="pointer-events: auto;"
      tabindex="-1"
    >
      <h2
        class="_title_e6bcc0"
        id="radix-:r1:"
      >
        This is important!
      </h2>
      <p
        class="_description_e6bcc0"
        id="radix-:r2:"
      >
        Are you sure about all of that stuff?
      </p>
      <footer>
        <button
          type="button"
        >
          Cancel
        </button>
        <button
          type="button"
        >
          Yes, do it!
        </button>
      </footer>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;
